contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -252,30 +267,29 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
     ) public onlyWhenUnlocked {
         require(currentValidatorSetSize() < validatorThreshold, "Validator set Threshold exceeded!");
         require(amount > minDeposit, "min deposit limit failed!");
-        require(heimdallFee >= minHeimdallFee, "Minimum amount is 1 Matic");
-
-        require(token.transferFrom(msg.sender, address(this), amount.add(heimdallFee)), "Transfer stake failed");
```

Comment:
with new `_topUpForFee` changes we need to do 2 transfer calls.

Reply:
Fixed